src/protocols/transaction.rs

```diff
@@ -60,13 +60,11 @@ impl Protocol for TxSubmissionProtocol {
     fn get_agency(&self) -> Agency {
         return match self.state {
             State::Idle => { Agency::Server }
-            State::TxIdsBlocking => {
-                // Typically, this would be client agency, but we pretend it's none since
-                // we just hang on to Client agency and never send any transactions.
-                Agency::None
```

Comment:
Is this then a failure case?

Reply:
This change was to more closely align with the spec, but still have our behavior of never sending any transactions to the server. In the spec, `State::TxIdsBlocking` is `Agency::Client`